src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3943,6 +3943,50 @@ GenTree* Lowering::TryLowerAndOpToAndNot(GenTreeOp* andNode)
     return andnNode;
 }
 
+//----------------------------------------------------------------------------------------------
+// Lowering::LowerBswapOp: Tries to contain GT_BSWAP node when possible
+//
+// Arguments:
+//    node - GT_BSWAP node to contain
+//
+// Notes:
+//    Containment is not performed when optimizations are disabled
+//    or when MOVBE instruction set is not found
+//
+void Lowering::LowerBswapOp(GenTreeOp* node)
+{
+    assert(node->OperIs(GT_BSWAP, GT_BSWAP16));
+
+    if (!comp->opts.OptimizationEnabled() || !comp->compOpportunisticallyDependsOn(InstructionSet_MOVBE))
+    {
+        return;
+    }
+
+    GenTree* operand = node->gtGetOp1();
+    if (IsContainableMemoryOp(operand) && IsSafeToContainMem(node, operand))
+    {
+        MakeSrcContained(node, operand);
+
+        // Return early so we don't do double-containing
+        return;
+    }
+
+    // Due to this instruction being rare we can perform TryGetUse here
+    // to simplify the pattern recognition required for GT_STOREIND
+    // In cases where TP matters, it should be performed inside the user
+    LIR::Use use;
+    if (!BlockRange().TryGetUse(node, &use))
+    {
+        return;
+    }
+
+    GenTree* user = use.User();
```

Comment:
The GT_BSWAP node is side effect-less so its not really required here. The node is transformed to GT_STOREIND(addr, contained BSWAP(x))

Reply:
The optimization can reorder the read and write of a local in the following case:
```scala
A = GT_LCL_VAR(V00)
B = GT_BSWAP(A)
C = GT_CNS_INT(10)
GT_STORE_LCL_VAR(V00, C)
w = GT_STORE_IND(<addr>, B)
```
It will produce e.g.
```asm
mov rax, 10
movbe [<addr>], rax
```
instead of
```asm
mov rcx, rax
bswap rcx
mov rax, 10
mov [<addr>], rcx
```
Today it is probably not easy/possible to come up with a case that results in this IR, but we should still handle this correctly (in particular, it might be possible to come up with something that trips over this due to address exposure). These kinds of cases is also why `IsSafeToContainMem` exists.

Reply:
Right, I did speak with Single about this one two weeks ago on Discord and he mentioned it would not be necessary. I will add it just to be safe then.